_rules/SC4-1-2-iframe-unique-name.md

```diff
@@ -0,0 +1,206 @@
+---
+name: IFrame has a unique accessible name
+
+description: |
+ Each iframe element has a unique accessible name
+
+success_criterion:
+- 4.1.2
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to `iframe` elements within a given [document context](#document-context) and [exposed to assistive technologies](#exposed-to-assistive-technologies) that have a [non-empty](#non-empty) accessible name.
+
+**Note:** 
+- Nested `iframe` contents treat the parent `iframe` element as the [document context](#document-context).
+
+### Expectation
+
+Each target element has an [accessible name](#accessible-name) that is unique within the [document context](#document-context).
```

Comment:
Suggestions:
- Allowing iframes with the same src to have the same accessible name
- Add an example of document context for clarification

Reply:
I don't think Wilco was suggesting to replace the Expectation. I'm not sure how iframes with the same source should be handled in the rule. My initial thought is that this is an applicability issue (if we follow the suggestion then they are not required to have unique accessible names). But if they are not required to have the same accessible name if they have the same src then it cannot be an expectation.

Reply:
To address the above two points.

1) I have added a note to the expectation, detailing `iframe` with the same `src` are allowed to have same `accessible name`.

2) Pass example 6, shows frames across multiple document contexts

Reply:
From the expectation I actually don't think it is clear whether it should just be unique across all iframe names, or across all accessible names (e.g. for images, landmarks etc.) across the page.

Reply:
Suggesting this Expectation, handling both my comment above and replacing our "document context" definition with "document tree" and "shadow tree" definitions from the DOM spec (working draft, I know): 

"Each target element has an [accessible name](#accessible-name) that is unique across all other accessible names for iframes within the same [document tree](https://www.w3.org/TR/dom41/#document-trees) or [shadow tree](https://www.w3.org/TR/dom41/#shadow-trees) as the target element."